From 9c4551c67d6fe854f7d37970398f5d5c9d17c28e Mon Sep 17 00:00:00 2001
From: raveit65 <mate@raveit.de>
Date: Thu, 29 Sep 2022 10:31:16 +0200
Subject: [PATCH] Revert "xutils: Get the correct PID for clients inside PID
 namespaces"

This reverts commit 07694559cc0c65ce1cca9ac33b165cef84c34d5e.

- fixing https://gitlab.gnome.org/GNOME/libwnck/-/issues/154#note_1562760
---
 libwnck/xutils.c | 36 +++---------------------------------
 1 file changed, 3 insertions(+), 33 deletions(-)

diff --git a/libwnck/xutils.c b/libwnck/xutils.c
index 60ae7b2..f2a2d3c 100644
--- a/libwnck/xutils.c
+++ b/libwnck/xutils.c
@@ -27,9 +27,6 @@
 #if HAVE_CAIRO_XLIB_XRENDER
 #include <cairo-xlib-xrender.h>
 #endif
-#ifdef HAVE_XRES
-#include <X11/extensions/XRes.h>
-#endif
 #include "screen.h"
 #include "window.h"
 #include "private.h"
@@ -1149,41 +1146,14 @@ int
 _wnck_get_pid (Screen *screen,
                Window  xwindow)
 {
-  int pid = -1;
-
-#ifdef HAVE_XRES
-  XResClientIdSpec client_spec;
-  long client_id_count = 0;
-  XResClientIdValue *client_ids = NULL;
-
-  client_spec.client = xwindow;
-  client_spec.mask = XRES_CLIENT_ID_PID_MASK;
-
-  if (XResQueryClientIds (DisplayOfScreen (screen), 1, &client_spec,
-                          &client_id_count, &client_ids) == Success)
-    {
-      long i;
-
-      for (i = 0; i < client_id_count; i++)
-        {
-          pid = XResGetClientPid (&client_ids[i]);
-          if (pid != -1)
-            break;
-        }
-
-      XResClientIdsDestroy (client_id_count, client_ids);
-
-      if (pid != -1)
-        return pid;
-    }
-#endif
+  int val;
 
   if (!_wnck_get_cardinal (screen, xwindow,
                            _wnck_atom_get ("_NET_WM_PID"),
-                           &pid))
+                           &val))
     return 0;
   else
-    return pid;
+    return val;
 }
 
 char*
-- 
2.37.3


